### DISCUSSION TITLE
How i can connect a real mobile device on the local host ?

### DISCUSSION THREAD

**Discussion by regisx001** [pocketbase/pocketbase#600] [Q&A]
I am A flutter developer and when i run the server localy on my machine , it does't work work on my mobile !

`/flutter (16277): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: ClientException: {url: http://127.0.0.1:8090/api/collections/demo/records, isAbort: true, statusCode: 0, response: {}, originalError: Connection refused}
E/flutter (16277): #0      PocketBase.send
package:pocketbase/src/client.dart:154
E/flutter (16277): <asynchronous suspension>
E/flutter (16277): #1      HomePage.build.<anonymous closure>
package:app/…/Home/home_page.dart:27
E/flutter (16277): <asynchronous suspension>
E/flutter (16277):`

could anyone help me ?

**Comment by ganigeorgiev** [comment]
Are you using an emulator or a real device?

If you want to expose your localhost, you can use something like ngrok https://ngrok.com/ and use the generated address when initializing the PocketBase client in your flutter app.

**Reply by regisx001** [reply]
A real device !

**Reply by ganigeorgiev** [reply]
Then install ngrok (https://ngrok.com/download) and run it in anew terminal with the 8090 port. It will generate a public secure address that you'll be able to access from your device.

**Reply by regisx001** [reply]
ok i will try it ! 
thank you !

**Comment by FlixMa** [comment]
Ngrok routes the traffic to your local computer via the internet. However, what you probably want is just the access on your local network.
Because the "real device" doesn't live on your machine, the address URL you specified will not work `http://127.0.0.1:8090/api/collections/demo/records`.

You need to find your servers (i.e. the device on which pocketable runs) IP address and base the URL on that, e.g. `http://192.168.1.42:8090/api/collections/demo/records`.

**Reply by ganigeorgiev** [reply]
Yep, that's also another way to access your local network, but note that the above will work only if the device and your pc are connected to the same network (eg. to the same wifi) and there is no firewall that blocks incoming connection to your localhost.

**Reply by richardolan** [reply]
I am running pocketbase on Windows and it is impossible to access through my network, is something wrong?
 Example: http://192.168.1.42:8090/api/collections/demo/records

Error: This site cannot be reached192.168.2.155 connection refused.
Try:

Check the connection.
Check proxy and firewall.
ERR_CONNECTION_REFUSED

**Reply by tigawanna** [reply]
Do ipconfig in the terminal and use the ipv4 value like so 
```sh
.\pocketbase.exe serve  your-ipv4-address
```
Ex
```sh
.\pocketbase.exe serve 192.168.20.87
```
[Article](https://dev.to/tigawanna/pocketbase-with-react-and-react-query-40dn)

**Reply by ganigeorgiev** [reply]
@tigawanna I don't think this is correct because the IP will not have any effect on the executable. You need to use the `--http=YOUR_IP:PORT` flag if want to provide custom address and port.

**Reply by tigawanna** [reply]
My bad , yes the --http= flag is definitely required

**Reply by richardolan** [reply]
resolved, thanks.
